test/test_checks.py

```diff
@@ -21,24 +19,53 @@ def test_get_noqa_lines(self):
         self.assertEqual(checker.get_noqa_lines(checker.get_file_contents()), [2])
 
 
-class TestFlake8Stdin(TestCase):
-    @expectedFailureIf(FLAKE8_VERSION == '3')
+class TestFlake8(TestCase):
+    @classmethod
+    def setUpClass(cls):
+        if version_info < (3, 0):
+            cls.expected_error_suffix = b'25: Q000 Remove bad quotes.'
+            # For some reason using "--select=Q" did suppress all outputs, so
+            # the result might contain non flake_quotes related errors
+            parameters = []
+        else:
+            cls.expected_error_suffix = b'24: Q000 Remove bad quotes.'
+            # Conversely Flake8 3.x currently has a bug that it doesn't report
+            # any error unless selected specifically
+            # See also: https://gitlab.com/pycqa/flake8/issues/183
+            parameters = ['--select=Q']
+        cls.command = ['flake8'] + parameters
+
+    def test_file(self):
+        """Test using stdin."""
+        filepath = get_absolute_path('data/doubles.py')
+        p = subprocess.Popen(self.command + [filepath],
+                             stdout=subprocess.PIPE, stderr=subprocess.PIPE)
```

Comment:
Right but if `flake8` breaks, then it's likely a problem out of our reach and implementing a workaround is likely silly. This level of testing is usually only if we encounter a regression, otherwise we have verified it works manually so it's good enough for now.